### PR TITLE
Add documentation about formatters and linters to README.dev.md

### PR DESCRIPTION
**Pull request by abelsiqueira**
**Description**

Update developer documentation with information to run formatter and linter locally.

**Related issues**:

- #32
- Blocked by #59 

**Instructions to review the pull request**

- Check that CHANGELOG.md has been updated if necessary
- Wait for #59 to be merged
- Follow instructions in the documentation and see if they are clear enough.

### CODE REVIEW

File: README.dev.md [hunk 1]
```diff
@@ -16,12 +16,67 @@ cd CUDA-wrappers
...
-:construction: See issues #7, #31, #32
+:construction: See issues #7, #31
+
+#### Linters and Formatters
+
+We use the following linters and formatters in this project:
+
+- [clang-format](clang.llvm.org/docs/ClangFormat.html)
+- [clang-tidy](clang.llvm.org/extra/clang-tidy)
+- [cmake-format and cmake-lint](cmake-format.readthedocs.io/en/latest/installation.html)
```
Comment (jspaaks) [inline README.dev.md:27]:
```suggestion
- [clang-format](https://clang.llvm.org/docs/ClangFormat.html)
- [clang-tidy](https://clang.llvm.org/extra/clang-tidy)
- [cmake-format and cmake-lint](https://cmake-format.readthedocs.io/en/latest/installation.html)
```

File: README.dev.md [hunk 2]
```diff
@@ -16,12 +16,67 @@ cd CUDA-wrappers
...
+- [cppcheck](https://cppcheck.sourceforge.io)
+- [flawfinder](https://dwheeler.com/flawfinder/)
+
+The formatter `clang-format` will format all source files, and `cmake-format` will format all CMake-related files.
+The linters will check for errors and bugs, but also style-related issues. So run the formatters before running the linters.
+
+Check how to run them in the [Building](#building).
```
Comment (jspaaks) [inline README.dev.md:34]:
```suggestion
Check how to run them in the [Building](#building) section below.
```

File: README.dev.md [hunk 3]
```diff
@@ -16,12 +16,67 @@ cd CUDA-wrappers
...
+
+In addition, you can install VSCode extensions for many of these linters. Here is a short list:
+
+- [VSCode extension for cmake-format](marketplace.visualstudio.com/items?itemName=cheshirekow.cmake-format)
+- [VSCode extension clang-tidy](marketplace.visualstudio.com/items?itemName=notskm.clang-tidy)
+- [VSCode extension Clang-Format](marketplace.visualstudio.com/items?itemName=xaver.clang-format)
```
Comment (jspaaks) [inline README.dev.md:40]:
```suggestion
- [VSCode extension for cmake-format](https://marketplace.visualstudio.com/items?itemName=cheshirekow.cmake-format)
- [VSCode extension clang-tidy](https://marketplace.visualstudio.com/items?itemName=notskm.clang-tidy)
- [VSCode extension Clang-Format](https://marketplace.visualstudio.com/items?itemName=xaver.clang-format)
```

Comment (abelsiqueira) [reply README.dev.md]: Thanks for the suggestions.

File: README.dev.md [hunk 4]
```diff
@@ -16,12 +16,67 @@ cd CUDA-wrappers
...
+
+where `<tool>` can be any of the following:
+
+- `cmake-format`
+- `cmake-lint`
+- `clang-format`
+- `clang-tidy`
+- `cppcheck`
+- `flawfinder`
+
```
Comment (jspaaks) [inline README.dev.md:79]: we could put the things you can get from pypi into a requirements.txt

Comment (abelsiqueira) [reply README.dev.md:91]: I have created #90 with that suggestion.

**Comment by abelsiqueira** [comment]
This branch has been updated, it should be possible to follow the commands now.

**Review by fdiblen — APPROVED** [review]
LGTM. :100:

**Comment by abelsiqueira** [comment]
Thanks for the review :100:⁰

**Pull request merged**